### PR TITLE
[Merged by Bors] - feat port/CategoryTheory.Yoneda

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2295
- [x] depends on: #2298

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2295
* leanprover-community/mathlib4#2298
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2295~~
* ~~leanprover-community/mathlib4#2298~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Yoneda.lean [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+  (Representable.has_representation : ∃ (_ : _)(_ : _ ⟶ F), _).choose
+set_option linter.uppercaseLean3 false 
+#align category_theory.functor.repr_X CategoryTheory.Functor.reprX
+
+/-- The (forward direction of the) isomorphism witnessing `F` is representable. -/
+noncomputable def reprF : yoneda.obj F.reprX ⟶ F :=
+  Representable.has_representation.choose_spec.choose
+#align category_theory.functor.repr_f CategoryTheory.Functor.reprF
+
+/- warning: category_theory.functor.repr_x clashes with
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Yoneda.lean:220]: I think you can remove this comment. It no longer applies, right?

File: Mathlib/CategoryTheory/Yoneda.lean [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+  (Corepresentable.has_corepresentation : ∃ (_ : _)(_ : _ ⟶ F), _).choose.unop
+set_option linter.uppercaseLean3 false 
+#align category_theory.functor.corepr_X CategoryTheory.Functor.coreprX
+
+/-- The (forward direction of the) isomorphism witnessing `F` is corepresentable. -/
+noncomputable def coreprF : coyoneda.obj (op F.coreprX) ⟶ F :=
+  Corepresentable.has_corepresentation.choose_spec.choose
+#align category_theory.functor.corepr_f CategoryTheory.Functor.coreprF
+
+/- warning: category_theory.functor.corepr_x clashes with
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Yoneda.lean:287]: Idem

File: Mathlib/CategoryTheory/Yoneda.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Yoneda.lean]: Removed

File: Mathlib/CategoryTheory/Yoneda.lean (hunk 2 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Yoneda.lean]: Removed

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**